### ISSUE TITLE
`extract_msg.signed_attachment.save` returning Message object after data has been written to disk

### ISSUE THREAD

**Issue by cccs-rs** [TeamMsgExtractor/msg-extractor#372]
**Bug Metadata**
* Version of extract_msg: [0.41.5]
* Your python version: Python [3.9.15]
* How did you launch extract_msg?
  - [ ] My command line or
  - [x] I used the extract_msg package

**Describe the bug**
Is there a reason why `extract_msg.signed_attachment.save()` returns back the Message object instead of null or the path to the written file? 
https://github.com/TeamMsgExtractor/msg-extractor/blob/3cffc2e0d82a0301cfaca2b05c5ef35bfc96a8cb/extract_msg/signed_attachment.py#L167-L179

**What code did you use or can we use to reproduce this error?
```python
            try:
                ret_value = attachment.save(customPath=self.working_directory,
                                            customFilename=customFilename, extractEmbedded=True)
            except Exception:
                continue

            if isinstance(attachment, extract_msg.signed_attachment.SignedAttachment):
                attachment_name = os.path.basename(ret_value)
                attachment_path = ret_value
```

**Traceback**
```
  File "/var/lib/assemblyline/.local/lib/python3.9/site-packages/assemblyline_v4_service/common/base.py", line 169, in handle_task
    self.execute(request)
  File "/opt/al_service/emlparser/emlparser.py", line 50, in execute
    self.handle_outlook(request)
  File "/opt/al_service/emlparser/emlparser.py", line 167, in handle_outlook
    attachment_name = os.path.basename(ret_value)
  File "/usr/local/lib/python3.9/posixpath.py", line 142, in basename
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not MessageSigned
```

**Additional context**
`self.type == AttachmentType.SIGNED`

**Comment by TheElementalOfDestruction** [comment]
I thought I fixed this issue when I was messing with attachments in the upcoming version, but looks like I missed it. It's supposed to return the child MSGFile instance if it's save method will end up being called to try to indicate that a child object was saved specially, but it was not doing that in several different ways. I've updated the typing to indicate this, although it may end up changing in the future or even before I actually get the next version out. 

Saving the embedded message as a file though was supposed to return the path as well. This has been fixed in the most recent commit, but I will leave this open until the next version is out. Your arguments you are providing should mean that, with the fixed and updated system, you should always be getting a path.

**Comment by TheElementalOfDestruction** [comment]
You know what, I think since I'm already changing the external API in a bunch of ways, I might just go one further and change how the save functions return. I feel like it is more natural for them to always return where they save to, especially since other factors can cause the name to get a bit mangled. I'll let you know the final result before I publish the next version, just so it is better documented.

**Comment by TheElementalOfDestruction** [comment]
This has now been implemented in the version 0.42.0. I ended up changing how the returns of all the save functions work, so be sure to check the new format so you aren't caught off guard by the different return types. It's now much more informative about what the function did so you can check it with code.

**Comment by cccs-rs** [comment]
Fantastic! Thank you so much! 😀
